utils_cv/detection/plot.py

```diff
@@ -0,0 +1,279 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""
+Helper module for visualizations
+"""
+from typing import List, Union, Tuple, Callable, Any
+from pathlib import Path
+
+import cv2
+import numpy as np
+import matplotlib.pyplot as plt
+from PIL.JpegImagePlugin import JpegImageFile
+
+from .references.coco_eval import CocoEvaluator
+
+
+class BoundingBoxParams:
+    """ Simple class to contain bounding box params. """
+
+    def __init__(
+        self,
+        rect_th: int = 2,
+        rect_color: Tuple[int, int, int] = (255, 0, 0),
+        text_size: float = 1,
+        text_th: int = 2,
+        text_color: Tuple[int, int, int] = (255, 255, 255),
+    ):
+        self.rect_th, self.rect_color, self.text_size, self.text_th, self.text_color = (
+            rect_th,
+            rect_color,
+            text_size,
+            text_th,
+            text_color,
+        )
+
+
+def plot_boxes(
+    im: JpegImageFile,
+    boxes: List[List[int]],
+    categories: List[str],
+    bbox_params: BoundingBoxParams = BoundingBoxParams(),
+) -> JpegImageFile:
+    """ Plot boxes on Image and return the Image
+
+    Args:
+        im: The image to plot boxes on
+        boxes: the boxes to plot - a list of [xmin, ymin, xmax, ymax] boxes
+        categories: list of catogires to label the boxes with
+        bbox_params: the parameter of the bounding boxes
+
+    Returns:
+        The same image with boxes and labels plotted on it
+    """
+    # Convert to RGB
+    im = cv2.cvtColor(im, cv2.COLOR_BGR2RGB)
+    if len(boxes) > 0:
+        for box, category in zip(boxes, categories):
+
+            # reformat boxes to be consumable by cv2
+            box = [(box[0], box[1]), (box[2], box[3])]
+
+            # Draw Rectangle with the coordinates
+            cv2.rectangle(
+                im,
+                box[0],
+                box[1],
+                color=bbox_params.rect_color,
+                thickness=bbox_params.rect_th,
+            )
+
+            # Write the prediction class
+            cv2.putText(
+                im,
+                category,
+                (box[0][0], box[0][1] + 15),
+                cv2.FONT_HERSHEY_SIMPLEX,
+                bbox_params.text_size,
+                color=bbox_params.text_color,
+                thickness=bbox_params.text_th,
+            )
+
+    return im
+
+
+def display_bounding_boxes(
```

Comment:
Nice example where we'd just have to pass in a list of BBox objects, and each BBox contains all the necessary information